src/cli/lint/autofix.ts

```diff
@@ -0,0 +1,108 @@
+import simpleGit from 'simple-git';
+
+import * as Git from '../../api/git';
+import { runESLint } from '../../cli/adapter/eslint';
+import { runPrettier } from '../../cli/adapter/prettier';
+import { isCiEnv } from '../../utils/env';
+import { createLogger, log } from '../../utils/logging';
+
+import type { Input } from './types';
+
+const AUTOFIX_COMMIT_MESSAGE = 'Run `skuba format`';
+
+const shouldPush = async ({
+  currentBranch,
+  dir,
+}: {
+  currentBranch?: string;
+  dir: string;
+}) => {
+  if (!isCiEnv()) {
+    // We're not running in a CI environment so we don't need to push autofixes.
+    // Ideally we'd drive this off of repository write permissions, but that is
+    // non-trivial to infer without attempting an actual write.
+    return false;
+  }
+
+  const isDefaultBuildkiteBranch =
+    currentBranch &&
+    [process.env.BUILDKITE_PIPELINE_DEFAULT_BRANCH, 'master', 'main'].includes(
+      currentBranch,
+    );
+
+  const isProtectedGitHubBranch = process.env.GITHUB_REF_PROTECTED === 'true';
+
+  if (isDefaultBuildkiteBranch || isProtectedGitHubBranch) {
+    // The current branch is a protected branch.
+    // We respect GitHub Flow; avoid pushing directly to the default branch.
+    return false;
+  }
+
+  let headCommitMessage;
+  try {
+    headCommitMessage = await Git.getHeadCommitMessage({ dir });
+  } catch {}
```

Comment:
Do we want to still commit if we can't find a head commit? Could that be potentially problematic eg. If our commit is the latest commit but it simply can't find it?